rust/src/dataset/scanner.rs

```diff
@@ -350,20 +354,63 @@ impl Scanner {
         Ok(Arc::new(schema))
     }
 
-    /// Create a stream from the Scanner.
-    pub async fn try_into_stream(&self) -> Result<DatasetRecordBatchStream> {
-        let plan = self.create_plan().await?;
-
+    fn execute_plan(&self, plan: Arc<dyn ExecutionPlan>) -> Result<SendableRecordBatchStream> {
         let session_config = SessionConfig::new();
         let runtime_config = RuntimeConfig::new();
         let runtime_env = Arc::new(RuntimeEnv::new(runtime_config)?);
         let session_state = SessionState::with_config_rt(session_config, runtime_env);
         // NOTE: we are only executing the first partition here. Therefore, if
         // the plan has more than one partition, we will be missing data.
         assert_eq!(plan.output_partitioning().partition_count(), 1);
-        Ok(DatasetRecordBatchStream::new(
-            plan.execute(0, session_state.task_ctx())?,
-        ))
+        Ok(plan.execute(0, session_state.task_ctx())?)
+    }
+
+    /// Create a stream from the Scanner.
+    pub async fn try_into_stream(&self) -> Result<DatasetRecordBatchStream> {
+        let plan = self.create_plan().await?;
+        Ok(DatasetRecordBatchStream::new(self.execute_plan(plan)?))
+    }
+
+    /// Scan and return the number of matching rows
+    pub async fn count_rows(&self) -> Result<usize> {
```

Comment:
is there plan to support `count_rows(filter)` ?

Reply:
This `count_rows` method is on the scanner, which may have a filter already specified.  So you can do:

```
            dataset
                .scan()
                .filter("filter_me > 15")
                .unwrap()
                .count_rows()
```